src/radical/pilot/utils/component.py

```diff
@@ -333,88 +416,96 @@ def __init__(self, cfg, session):
       #                            scope='entity',
       #                            start='get',
       #                            stop=['put', 'drop'])
+        self._prof.prof('init1', uid=self._uid, msg=self._prof.path)
 
         self._q    = None
         self._in   = None
         self._out  = None
         self._poll = None
         self._ctx  = None
 
-        # initialize the Process base class for later fork.
-        super(Component, self).__init__(name=self._uid, log=self._log)
+        self._thread = None
+        self._term   = mt.Event()
+
+
+    # --------------------------------------------------------------------------
+    #
+    def start(self):
 
-        # make sure we bootstrapped ok
-        self.is_valid()
-        self._session._to_stop.append(self)
+        sync = mt.Event()
+        self._thread = mt.Thread(target=self._worker_thread, args=[sync])
+        self._thread.daemon = True
+        self._thread.start()
 
+        # TODO: add timeout to shield against failing initlization?
```

Comment:
This does not look 'clean' and I saw it in other places. Should we have a way to pub/sub component initialization? Maybe related to the TODO of line 465?

Reply:
Ah, thanks, I should remove this comment - that shieding is now in place via this `sync` event you see above, which, well, syncs startup and initialization.

Reply:
I am not sure if this is really needed, the heartbeat startup timeout should kick in anyway.  I left it as an open question to revisit eventually in case we run into troubles with faulty init routines.  If you think this needs resolving let me know, and I'll dig through it more thoroughly!

Reply:
The PR is the first of its kind (bye bye darn mt/mp wrapper, yay!) so let's keep it like that and see how it works. I would maybe add something to the TODO about using different methods than a wait to manage state transitions (in this case transition from initialize to initialized)?

Reply:
Ack, my thoughts as well.  I'll expand that comment.

Reply:
Ah well, I just fixed it - was shorter than adding the comment :-D